components/tidb_qe/src/error.rs

```diff
@@ -0,0 +1,116 @@
+// Copyright 2019 TiKV Project Authors. Licensed under Apache-2.0.
+
+use crate::codec::Error as CodecError;
+
+#[derive(Fail, Debug)]
+pub enum EvaluateError {
+    #[fail(display = "Execution terminated due to exceeding max time limit")]
+    MaxExecuteTimeExceeded,
+
+    /// This variant is only a compatible layer for existing CodecError.
+    /// Ideally each error kind should occupy an enum variant.
+    #[fail(display = "{}", msg)]
+    Custom { code: i32, msg: String },
+
+    #[fail(display = "{}", _0)]
+    Unknown(String),
+}
+
+impl EvaluateError {
+    /// Returns the error code.
+    pub fn code(&self) -> i32 {
+        match self {
+            // TODO: We should assign our own error code
+            EvaluateError::MaxExecuteTimeExceeded => 1105,
```

Comment:
I think we should organize all error codes in an individual module and assign a constant for them.

Reply:
This is the only place need these codes so looks like separating codes does not help readability, flexibility or reduce repeats.